pkg/agent/datamodel/types.go

```diff
@@ -359,39 +359,129 @@ type LinuxProfile struct {
+	CloudProviderBackoffMode          string                `json:"cloudProviderBackoffMode"`
+	CloudProviderBackoff              *bool                 `json:"cloudProviderBackoff,omitempty"`
+	CloudProviderBackoffRetries       int                   `json:"cloudProviderBackoffRetries,omitempty"`
+	CloudProviderBackoffJitter        float64               `json:"cloudProviderBackoffJitter,omitempty"`
+	CloudProviderBackoffDuration      int                   `json:"cloudProviderBackoffDuration,omitempty"`
+	CloudProviderBackoffExponent      float64               `json:"cloudProviderBackoffExponent,omitempty"`
+	CloudProviderRateLimit            *bool                 `json:"cloudProviderRateLimit,omitempty"`
+	CloudProviderRateLimitQPS         float64               `json:"cloudProviderRateLimitQPS,omitempty"`
+	CloudProviderRateLimitQPSWrite    float64               `json:"cloudProviderRateLimitQPSWrite,omitempty"`
+	CloudProviderRateLimitBucket      int                   `json:"cloudProviderRateLimitBucket,omitempty"`
+	CloudProviderRateLimitBucketWrite int                   `json:"cloudProviderRateLimitBucketWrite,omitempty"`
+	CloudProviderDisableOutboundSNAT  *bool                 `json:"cloudProviderDisableOutboundSNAT,omitempty"`
+	NonMasqueradeCidr                 string                `json:"nonMasqueradeCidr,omitempty"`
+	NodeStatusUpdateFrequency         string                `json:"nodeStatusUpdateFrequency,omitempty"`
+	HardEvictionThreshold             string                `json:"hardEvictionThreshold,omitempty"`
+	CtrlMgrNodeMonitorGracePeriod     string                `json:"ctrlMgrNodeMonitorGracePeriod,omitempty"`
+	CtrlMgrPodEvictionTimeout         string                `json:"ctrlMgrPodEvictionTimeout,omitempty"`
+	CtrlMgrRouteReconciliationPeriod  string                `json:"ctrlMgrRouteReconciliationPeriod,omitempty"`
+	LoadBalancerSku                   string                `json:"loadBalancerSku,omitempty"`
+	ExcludeMasterFromStandardLB       *bool                 `json:"excludeMasterFromStandardLB,omitempty"`
+	AzureCNIVersion                   string                `json:"azureCNIVersion,omitempty"`
+	AzureCNIURLLinux                  string                `json:"azureCNIURLLinux,omitempty"`
+	AzureCNIURLWindows                string                `json:"azureCNIURLWindows,omitempty"`
+	KeyVaultSku                       string                `json:"keyVaultSku,omitempty"`
+	MaximumLoadBalancerRuleCount      int                   `json:"maximumLoadBalancerRuleCount,omitempty"`
+	ProxyMode                         api.KubeProxyMode     `json:"kubeProxyMode,omitempty"`
+	PrivateAzureRegistryServer        string                `json:"privateAzureRegistryServer,omitempty"`
+	OutboundRuleIdleTimeoutInMinutes  int32                 `json:"outboundRuleIdleTimeoutInMinutes,omitempty"`
+}
+
 // MasterProfile represents the definition of the master cluster
 type MasterProfile struct {
```

Comment:
we can remove this later

Reply:
Yes, I can remove the code not needed? Do you mean we can remove "PrivateAzureRegistryServer" and "OutboundRuleIdleTimeoutInMinutes" fields or the "MasterProfile" struct?